src/node_data.ts

```diff
@@ -114,13 +114,20 @@ function importSingleNode(node: Node, fallbackKey?: Key) {
   return data;
 }
 
+/**
+ * In lib.dom.d.ts in typescript v3.03 and before both Node.firstChild and Node.nextSibling were of type Node
+ * Starting in v3.1 firstChild is of type ChildNode, causing a type error.  Asserting that the type of child
+ * is NodeOrNull allows the code to compile under both versions.
+ */
+type NodeOrNull = Node | null;
+
 /**
  * Imports node and its subtree, initializing caches.
  */
 function importNode(node: Node) {
   importSingleNode(node);
 
-  for (let child = node.firstChild; child; child = child.nextSibling) {
+  for (let child:NodeOrNull = node.firstChild; child; child = child.nextSibling) {
```

Comment:
Throughout the code we have a space after `:` for types. This should be made consistent.

Reply:
Agreed.  I have made this change and also replaced the type NodeOrNulll with just Node|null as it is neater and consistent with the code style in the code base.